packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/sagemaker/create-training-job.ts

```diff
@@ -163,6 +163,14 @@ export class SageMakerCreateTrainingJob extends sfn.TaskStateBase implements iam
       throw new Error('Must define either an algorithm name or training image URI in the algorithm specification');
     }
 
+    // validate the algorithmName if the algorithmName is defined
```

Comment:
We need to add validation also for the case where [both `algorithmName` and `trainingImage` are specified](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_AlgorithmSpecification.html#:~:text=Note%20that%20the%20AlgorithmName%20parameter%20is%20mutually%20exclusive%20with%20the%20TrainingImage%20parameter.%20If%20you%20specify%20a%20value%20for%20the%20AlgorithmName%20parameter%2C%20you%20can%27t%20specify%20a%20value%20for%20TrainingImage%2C%20and%20vice%20versa.):
```
if (props.algorithmSpecification.algorithmName && props.algorithmSpecification.trainingImage) {
      throw ... 
}

Reply:
Oh, it's true. I will add it